### ISSUE TITLE
Navigation Block: "Existing menu" confusion as version in customizer is does not get updated to the page

### ISSUE THREAD

**Issue by aleone89** [Automattic/wp-calypso#52676]
### Steps to reproduce the behavior

1. Go to Pages > Add new > Add a title > page content.
2. Add navigation block > choose "Existing Menu" 
3. Choose a menu from the dropdown 
4. Menu gets added > Save changes 
5. Exit page 
6. Go to Appearance > Customize > Menus > go to chosen menu 
7. Change menu (add/remove menu items) > Publish changes 
8. Go to Pages > click on previously created page that contains menu

#### What I expected to happen

Changes made in the customizer to the menu to reflect on the page, as a existing menu was chosen

#### What actually happened

No changes were made

### Context

This came around during my virtual team meet up where we were going back to basics with the product. 

#### Browser / OS version

All

#### Is this specific to the applied theme? Which one?

All, in this Twenty Twenty was used. 

#### Does this happen on simple or atomic sites or both?

Both.

#### Is there any console output or error text?

N/A

#### Level of impact (Does it block purchases? Does it affect more than just one site?)

Low

#### Reproducibility (Consistent, Intermittent) Leave empty for consistent.

Consistent

**Comment by renata-franco** [comment]
The issue was reported on the ticket [4195704](4195704-zd-woothemes).

**Comment by kosiew** [comment]
Hi @aleone89 ,

Thanks for reporting this and the detailed steps.


From "Add Links to your Navigation" at https://wordpress.com/support/full-site-editing/fse-blocks/navigation-block/#add-links-to-your-navigation, "Existing Menu" is a way to initialize the Navigation block. It is possible to amend the Navigation block after initializing by adding further links.

In addition, from the Code Editor (here is a sample), there is no record of the origin of the menu items after initializing.

![image](https://user-images.githubusercontent.com/29057562/128693658-e4ad5b0b-d9e9-4cd0-8d8f-718a69a86890.png)

In other words, the "Existing Menu" option is a way to initialize the Navigation block, not to synchronize with the menu itself.

Triaged for further comment by others.

**Comment by dpasque** [comment]
Tagging @Automattic/flow-patrol-create based on issue content

**Comment by yansern** [comment]
This is marked as **Won't Fix** by @Automattic/flow-patrol-create because the Navigation Block is built to keep that page menu as block meta (as explained by @kosiew) instead of dynamically retrieve them. To fix this is to revamp Navigation Block to behave differently.

@aleone89 I suggest reporting this issue directly to Gutenberg https://github.com/WordPress/gutenberg/issues.

**Comment by simison** [comment]
cc @jasmussen in case there's any navigation block feedback to extract here.

**Comment by jasmussen** [comment]
The "Existing menu" is, as Yan noted, meant as a shortcut to populate the block where you create it, it's not meant to create a two-way link. As we explore enhancing the navigation block setup state, I will see what I can do to provide additional context here. Possibly along these lines:

<img width="740" alt="Setup states   appenders" src="https://user-images.githubusercontent.com/1204802/133220856-e437b5fc-cae2-4419-93e1-2a3729b7d8ad.png">

**Comment by annbingle** [comment]
I have a similar issue reported in zd-37118681. The theme is Appleton. We ended up updating the menu via the Navigation Block. However, the user still wants to use the Menu in Customizer as they find it easier.
